### PR TITLE
fix: create a recursive check for plugins and sensors

### PR DESCRIPTION
**Pull request by kapeps**
Depends on:
- [ ] https://github.com/rock-gazebo/simulation-orogen-rock_gazebo/pull/27
- [ ] https://github.com/tidewise/simulation-orogen-gazebo_usv/pull/7

### CODE REVIEW

File: bindings/ruby/lib/rock_gazebo/syskit/robot_definition_extension.rb [hunk 1]
```diff
@@ -361,22 +361,11 @@ def normalize_name(name)
 
             # @api private
             #
-            # Define devices for all links and sensors in the model
-            def load_gazebo_robot_model(
+            # Describes recursively all sensors and plugins in the model
+            def load_gazebo_robot_submodel(
```
Comment (jhonasiv) [inline bindings/ruby/lib/rock_gazebo/syskit/robot_definition_extension.rb]: I would have named it in the plural, `load_gazebo_robot_submodels`. Besides that, this looks good to me!

**Review by jhonasiv — APPROVED** [review]

**Comment by doudou** [comment]
Code looks good ... Just missing unit tests.

**Review by doudou — COMMENTED** [review]
Just needs unit tests ...

File: bindings/ruby/test/worlds/attached_model_with_submodel.world [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+                <name>included_model</name>
+                <uri>model://model_with_submodel</uri>
+            </include>
+        </model>
+
+        <model name="other_model">
+            <link name="l" />
+        </model>
+    </world>
+</sdf>
```
Comment (caioaamaral) [inline bindings/ruby/test/worlds/attached_model_with_submodel.world]: eof, are you using vscode with our configuration?

Comment (kapeps) [reply bindings/ruby/test/worlds/attached_model_with_submodel.world]: Done!

File: bindings/ruby/lib/rock_gazebo/syskit/robot_definition_extension.rb [hunk 3]
```diff
@@ -359,12 +366,52 @@ def normalize_name(name)
...
+                    )
+                end
+
+                sdf_model.each_model do |sdf_submodel|
+                    load_gazebo_robot_submodels(
+                        sdf_submodel,
+                        name: name + "_" + sdf_submodel.name,
+                        prefix_device_with_name: true,
+                        deployment_prefix: deployment_prefix + ":" + sdf_submodel.name,
+                        is_submodel: true
```
Comment (doudou) [inline bindings/ruby/lib/rock_gazebo/syskit/robot_definition_extension.rb]: Don't like this is_submodel here. We should be able to do without by changing the name generation logic to handle both cases, just like we did for the orogen model side.

**Review by doudou — APPROVED** [review]

**Review by rodrigoleonello — APPROVED** [review]

**Pull request merged**